### PR TITLE
CSCFAIRMETA-1033: File metadata validation bug

### PR DESCRIPTION
**Pull request by tahme**
* Fix errors in file metadata use_category validation
* Use <ValidationErrors> instead of <ValidationError> for array of errors
* Add tests to file and directory metadata forms
* Fix `referenceData.data.jsx` containing file types in place of use categories

### CODE REVIEW

File: etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  cancellation: true,
+})
+
+const Env = new EnvClass()
+const Accessibility = new AccessibilityClass(Env)
+const ElasticQuery = new ElasticQueryClass(Env)
+const Locale = new LocaleClass(Accessibility, ElasticQuery)
+
+const getStores = () => {
+  Env.Flags.setFlag('METAX_API_V2', true)
```
Comment (ismotee) [inline etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx:34]: not needed anymore

File: etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+import { SaveButton } from '../../../js/components/qvain/general/buttons'
+import { ValidationErrors } from '../../../js/components/qvain/general/errors/validationError'
+
+global.Promise = require('bluebird')
+
+Promise.config({
+  cancellation: true,
+})
+
+const Env = new EnvClass()
```
Comment (ismotee) [inline etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx:28]: nit: buildStores function from stores/stores could do these automatically.

File: etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+      await selectUseCategory('Documentation')
+      await expectSave({
+        title: 'testdir',
+        useCategory: 'http://uri.suomi.fi/codelist/fairdata/use_category/code/documentation',
+      })
+    })
+
+    it('should fail to save when use category is not set', async () => {
+      await render(true)
+      // await expectSave(undefined)
```
Comment (ismotee) [inline etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx:257]: clear commented code. Also three flushPromise calls in a row would deserve an explanation/comment

**Review by ismotee — CHANGES_REQUESTED** [review]

File: etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx (hunk 1 above)
Comment (tahme) [reply etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx]: removed

File: etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx (hunk 2 above)
Comment (tahme) [reply etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx]: true, this was originally copied from another test file. switched to buildStores

File: etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx (hunk 3 above)
Comment (tahme) [reply etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx]: those are from debugging, forgot to remove them (that problem was fixed by adding a wrapper.update to expectSave)

**Review by ismotee — APPROVED** [review]

**Pull request merged**